lib/foreman/renderer.rb

```diff
@@ -269,5 +299,99 @@ def pxe_config
       @kernel = @host.operatingsystem.kernel @host.arch
       @initrd = @host.operatingsystem.initrd @host.arch
     end
+
+    # registeres host to be observed for call of deprecated methods
+    def register_method_deprecations_for(host)
+      return if !@host || !@host.respond_to?(:info) || @host.respond_to?(:monitored_for_deprecations)
+      host.class_eval { include MethodDeprecations }
+    end
+
+    # loggs the deprecation messages
+    def log_method_deprecations
+      return unless Thread.current[:render_deprecations]
+      warn = "DEPRECATION WARNING:"
+      msgs = {
+        :info => "call of @host.info detected in #{template_name}. Use host_enc instead.",
+        :params => "call of @host.params['some_param'] detected in #{template_name}. Use host_param('some_param') instead."
+      }
+
+      methods = %i(param_true? param_false?)
+      methods.each do |m|
+        msgs[m] = "call of @host.#{m}('some_param') detected in #{template_name}. Use host_#{m}('some_param') instead."
+      end
+
+      Thread.current[:render_deprecations].each do |depr_method|
+        logger.info "#{warn} #{msgs[depr_method]}"
+        Thread.current[:render_deprecations].shift
+      end
+    end
+
+    # monitors call of deprecated methods on @host
+    module MethodDeprecations
+      METHODS = %i(info params)
+      P_METHODS = %i(param_true? param_false?)
+      ALL_METHODS = METHODS + P_METHODS
+
+      private
+
+      def stash_to_thread(method_name)
+        if Thread.current[:render_deprecations].nil?
+          Thread.current[:render_deprecations] = [method_name]
+        else
+          Thread.current[:render_deprecations] << method_name
+        end
+      end
+
+      class << self
+        def changed_method_name(method_name, changed_name)
+          "#{method_name.to_s.chop}_#{changed_name}?"
+        end
+
+        def define_param_bool_methods(new_name, old_name)
+          P_METHODS.each do |m|
+            new_method_name = changed_method_name(m, new_name)
+            old_method_name = changed_method_name(m, old_name)
+
+            define_method new_method_name do |host|
+              stash_to_thread m
+              self.send(old_method_name, host)
+            end
+          end
+        end
+
+        def define_methods_without_args(new_name, old_name)
+          METHODS.each do |m|
+            define_method "#{m}_#{new_name}" do
```

Comment:
> Moreover, it would take note each time info method was called (like here or here), but we do not want to deprecate method itself, only its usage in templates.

My earlier suggestion was to use a thread variable for this - set a thread variable in the renderer, unset it in an `ensure` block, and then the hypothetical `renderer_deprecation` method in the above example can check if the variable is present.